### PR TITLE
Fixed issue with filtering task based on status

### PR DESCRIPTION
**Pull request by ice-rahul**
[issue 73](https://github.com/Real-Dev-Squad/website-status/issues/73) 

issue_73

PR to fix the status issue

### CODE REVIEW

**Review by swarajpure — DISMISSED** [review]
Less time complexity, less hardcoding, removed unnecessary re-renders.

Great job 🥳 
Approving from my side.

**Comment by swarajpure** [comment]
Congratulations on your first PR in open source and that too in Real Dev Squad 🥳 🎉

**Comment by ice-rahul** [comment]
> Congratulations on your first PR in open source and that too in Real Dev Squad 🥳 🎉

thanks :)

**Review by harshith-venkatesh — DISMISSED** [review]

File: src/pages/index.tsx [hunk 1]
```diff
@@ -9,33 +9,28 @@ import Accordion from '@/components/Accordion';
 
 const TASKS_URL = `${process.env.NEXT_PUBLIC_BASE_URL}/tasks`;
 
-const renderCardList = (tasks: task[]) => tasks.map(
-  (item: task) => <Card content={item} key={item.id} />,
-);
+function renderCardList(tasks: task[]) {
```
Comment (harshith-venkatesh) [inline src/pages/index.tsx:12]: Any reason for shifting away from arrow function approach

Comment (harshith-venkatesh) [reply src/pages/index.tsx:12]: Rest all LGTM

File: package.json [hunk 2]
```diff
@@ -45,7 +45,7 @@
     "eslint-plugin-prettier": "^3.3.1",
     "eslint-plugin-react": "^7.22.0",
     "eslint-plugin-react-hooks": "^4.2.0",
-    "husky": "^4.3.8",
+    "husky": "^5.0.9",
```
Comment (whyDontI) [inline package.json:48]: Let's not push the code which is not required for the current issue we are solving.
But as this is a small change and not affecting much, we can keep this change for now.

Let's keep this in mind from future 😃

**Review by whyDontI — CHANGES_REQUESTED** [review]
Nice work 🎉 
Just we need to handle the case when there are no tasks in the response.
Show some error message on the page.

Rest all looks good 👍🏽

File: src/pages/index.tsx (hunk 1 above)
Comment (ice-rahul) [reply src/pages/index.tsx:12]: If we would be doing it the other way around the function expression will become a one-liner as per ESLINT standards.
Also, that one-liner will be of length 106 characters. But, ESLINT allows only 100 characters in a line
So changed it to function declaration.

File: package.json (hunk 2 above)
Comment (ice-rahul) [reply package.json:48]: Actually, I was facing an issue while committing it.
So in order to fix the commit issue, I had to update it

File: src/pages/index.tsx (hunk 1 above)
Comment (harshith-venkatesh) [reply src/pages/index.tsx:12]: Hmmmm, I mean changing the way just for eslint issue is something really new to me.
But ok, you should have thought a lot before coming to this conclusion so cool

**Review by harshith-venkatesh — DISMISSED** [review]

File: package.json (hunk 2 above)
Comment (whyDontI) [reply package.json:48]: What issue, can you please explain here?

Comment (ice-rahul) [reply package.json:48]: it was related to the pre-commit hook

**pre-commit hook failed (add --no-verify to bypass)**

[https://github.com/typicode/husky/issues/716](https://github.com/typicode/husky/issues/716)

File: src/pages/index.tsx [hunk 3]
```diff
@@ -44,24 +39,21 @@ const Index: FC = () => {
...
-                <Accordion title="Active">
-                  {renderCardList(activeTasks)}
-                </Accordion>
-                <Accordion title="Completed">
-                  {renderCardList(completeTasks)}
-                </Accordion>
+                {
+                  Object.keys(filteredTask).length > 0
+                    ? Object.keys(filteredTask).map((key) => (
+                      <Accordion title={key}>
```
Comment (sumitd94) [inline src/pages/index.tsx:52]: `key` is missing here, can we please add it.

**Review by sumitd94 — CHANGES_REQUESTED** [review]
One small change

File: src/pages/index.tsx (hunk 3 above)
Comment (ice-rahul) [reply src/pages/index.tsx:52]: Do we really need to add it?
As ESLINT is not having issues also we are having a unique title prop attached to Accordion.

Comment (whyDontI) [reply src/pages/index.tsx:52]: https://reactjs.org/docs/lists-and-keys.html

**Review by whyDontI — APPROVED** [review]
Nice work 👍🏽

File: src/pages/index.tsx (hunk 3 above)
Comment (ice-rahul) [reply src/pages/index.tsx]: > https://reactjs.org/docs/lists-and-keys.html
> @ice-rahul You might want to go through this.

thanks for the reference 👍

**Pull request merged**